src/sentry/integrations/jira/models/create_issue_metadata.py

```diff
@@ -0,0 +1,161 @@
+    """
+    The very long custom field name corresponding to some namespace, plugin,
+    and custom field name.
+    """
+    custom_id: str | None = None
+    """
+    A unique identifier for a field on an issue, in the form of 'customfield_<int>'
+    """
+    system: str | None = None
+    """
+    TODO(Gabe): Figure out what this is used for
+    """
+    items: str | None = None
+    """
+    Specifies the subtype for aggregate type, such as `array`. For any
+    non-aggregate types, this will be None
+    """
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> "JiraSchema":
+        schema_type = data["type"]
+        custom = data.get("custom")
+        custom_id = data.get("custom_id")
+        system = data.get("system")
+        items = data.get("items")
+
+        return cls(
+            schema_type=schema_type, custom=custom, custom_id=custom_id, system=system, items=items
+        )
+
+    @classmethod
+    def from_dict_list(cls, data: list[dict[str, Any]]) -> list["JiraSchema"]:
+        return [cls.from_dict(item) for item in data]
+
+
+@dataclass(frozen=True)
+class JiraField:
+    """
+    Represents a Jira Issue Field, which is queried directly via Jira's issue
+    creation metadata API:
+    https://developer.atlassian.com/server/jira/platform/rest/v10000/api-group-issue/#api-api-2-issue-createmeta-projectidorkey-issuetypes-issuetypeid-get
+    """
+
+    required: bool
+    schema: JiraSchema
+    name: str
+    key: str
+    operations: list[str]
+    has_default_value: bool
+
+    def is_custom_field(self) -> bool:
+        return self.schema.custom is not None
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> "JiraField":
+        required = data["required"]
+        name = data["name"]
+        key = data["key"]
+        operations = data["operations"]
+        has_default_value = data.get("hasDefaultValue") or False
+        raw_schema = data["schema"]
+
+        schema = JiraSchema.from_dict(raw_schema)
+
+        return cls(
+            required=required,
+            name=name,
+            key=key,
+            schema=schema,
+            operations=operations,
+            has_default_value=has_default_value,
+        )
+
+    @classmethod
+    def from_dict_list(cls, data: dict[str, dict[str, Any]]) -> dict[str, "JiraField"]:
+        return {key: cls.from_dict(val) for key, val in data.items()}
+
+    def get_field_type(self) -> JiraSchemaTypes | None:
+        type: str | None = self.schema.schema_type
+        if type == JiraSchemaTypes.array:
+            type = self.schema.items
+
+        if type not in JiraSchemaTypes:
+            return None
+
+        return JiraSchemaTypes(type)
+
+
+@dataclass(frozen=True)
+class JiraIssueTypeMetadata:
+    id: str
+    description: str
+    name: str
+    subtask: bool
+    icon_url: str
+    url: str
+    fields: dict[str, JiraField]
+
+    @classmethod
+    def from_dict(cls, data: dict[str, Any]) -> "JiraIssueTypeMetadata":
```

Comment:
Using this as a learning opportunity. Do type references have other benefits other than saying this type is only used here?

Reply:
Discussed this via slack, but as far as I know, this is just the older way of specifying forward refs pre python 3.7. Importing `__future__.annotations` and removing the quotes also works the same.